main/db.c

```diff
@@ -341,17 +340,17 @@ static int ast_db_rollback_transaction(void)
 
 int ast_db_put(const char *family, const char *key, const char *value)
 {
-	char fullkey[MAX_DB_FIELD];
+	char *fullkey;
 	size_t fullkey_len;
 	int res = 0;
 
-	if (strlen(family) + strlen(key) + 2 > sizeof(fullkey) - 1) {
-		ast_log(LOG_WARNING, "Family and key length must be less than %zu bytes\n", sizeof(fullkey) - 3);
+	fullkey_len = ast_asprintf(&fullkey, "/%s/%s", family, key);
+	if (fullkey_len < 0) {
```

Comment:
`fullkey_len` is a `size_t`

Reply:
I wonder why the compiler didn't catch that.